src/core/stream_set.c

```diff
@@ -75,28 +100,29 @@ QuicStreamSetTraceRundown(
     _In_ QUIC_STREAM_SET* StreamSet
     )
 {
-    if (StreamSet->StreamTable == NULL) {
-        return; // No streams have been created yet.
+    if (StreamSet->StreamTable != NULL) {
+        CXPLAT_HASHTABLE_ENUMERATOR Enumerator;
+        CXPLAT_HASHTABLE_ENTRY* Entry;
+        CxPlatHashtableEnumerateBegin(StreamSet->StreamTable, &Enumerator);
+        while ((Entry = CxPlatHashtableEnumerateNext(StreamSet->StreamTable, &Enumerator)) != NULL) {
+            QuicStreamTraceRundown(
+                CXPLAT_CONTAINING_RECORD(Entry, QUIC_STREAM, TableEntry));
+        }
+        CxPlatHashtableEnumerateEnd(StreamSet->StreamTable, &Enumerator);
     }
 
-    CXPLAT_HASHTABLE_ENUMERATOR Enumerator;
-    CXPLAT_HASHTABLE_ENTRY* Entry;
-    CxPlatHashtableEnumerateBegin(StreamSet->StreamTable, &Enumerator);
-    while ((Entry = CxPlatHashtableEnumerateNext(StreamSet->StreamTable, &Enumerator)) != NULL) {
+    for (CXPLAT_LIST_ENTRY *Link = StreamSet->WaitingStreams.Flink;
+         Link != &StreamSet->WaitingStreams;
+         Link = Link->Flink) {
         QuicStreamTraceRundown(
-            CXPLAT_CONTAINING_RECORD(Entry, QUIC_STREAM, TableEntry));
+            CXPLAT_CONTAINING_RECORD(Link, QUIC_STREAM, WaitingLink));
     }
-    CxPlatHashtableEnumerateEnd(StreamSet->StreamTable, &Enumerator);
 }
 
 _IRQL_requires_max_(DISPATCH_LEVEL)
 _Success_(return != FALSE)
 BOOLEAN
-QuicStreamSetInsertStream(
-    _Inout_ QUIC_STREAM_SET* StreamSet,
-    _In_ QUIC_STREAM* Stream
-    )
-{
+QuicStreamSetLazyInitStreamTable(_Inout_ QUIC_STREAM_SET* StreamSet) {
```

Comment:
```suggestion
QuicStreamSetLazyInitStreamTable(
    _Inout_ QUIC_STREAM_SET* StreamSet
    )
{
```

In the product code we generally always use this format.